### PR TITLE
Change file permissions on console serving cert

### PR DESCRIPTION
**Pull request by praveenkumar**
Have a look to https://github.com/openshift/origin-web-console-server/issues/37 and https://github.com/openshift/openshift-ansible/pull/8558/files , This will also solve minishift issue https://github.com/minishift/minishift/issues/2809

### CODE REVIEW

**Comment by praveenkumar** [comment]
cc @deads2k @mfojtik @smarterclayton

**Comment by kameshsampath** [comment]
+1

File: install/openshift-web-console-operator/install.yaml [hunk 1]
```diff
@@ -77,12 +77,12 @@ objects:
         volumes:
         - name: serving-cert
           secret:
-            defaultMode: 400
+            defaultMode: 0440
```
Comment (deads2k) [inline install/openshift-web-console-operator/install.yaml]: Just remove the tag entirely.  This doesn't work well.

Comment (praveenkumar) [reply install/openshift-web-console-operator/install.yaml]: @deads2k you mean this `serving-cert` one?

**Comment by praveenkumar** [comment]
@deads2k do we need any other changes or this is sufficient?

**Comment by praveenkumar** [comment]
ping @deads2k @mfojtik

**Comment by sspeiche** [comment]
+1 to getting this out. We want to roll out an updated CDK/minishift release that supports 3.11, since it has been released for a couple of weeks now. Getting a lot of requests for it.

File: install/openshift-web-console-operator/install.yaml [hunk 2]
```diff
@@ -77,12 +77,11 @@ objects:
         volumes:
         - name: serving-cert
           secret:
-            defaultMode: 400
             secretName: openshift-web-console-operator-serving-cert
             optional: true
         - name: config
           configMap:
-            defaultMode: 440
+            defaultMode: 0440
```
Comment (deads2k) [inline install/openshift-web-console-operator/install.yaml]: I'm pretty sure default modes don't work right, but ok.

**Comment by praveenkumar** [comment]
@jwforres Problem is this coming from the CI and I am not able to reproduce it local system :( , If @deads2k thinks we should remove this from configMap also then I better to update this PR only. Also this is against 3.11 branch only so no cherry pick needed.

**Comment by praveenkumar** [comment]
@jwforres @deads2k made required changes now, hope we are good here.

**Comment by jwforres** [comment]
@praveenkumar with your changes the console comes up ok locally?

**Comment by praveenkumar** [comment]
> with your changes the console comes up ok locally?

@jwforres looks like coming up as expected.

```console
✔ ~/work/github/practice/go/src/github.com/openshift/origin [pr/21250 L|✔] 
19:23 $ make
hack/build-go.sh  
++ Building go targets for linux/amd64: cmd/hypershift cmd/openshift cmd/oc cmd/oadm cmd/template-service-broker cmd/openshift-node-config vendor/k8s.io/kubernetes/cmd/hyperkube cmd/sdn-cni-plugin vendor/github.com/containernetworking/plugins/plugins/ipam/host-local vendor/github.com/containernetworking/plugins/plugins/main/loopback
[INFO] hack/build-go.sh exited with code 0 after 00h 01m 52s

$ cd _output/local/bin/linux/amd64/
$ ./oc version
oc v3.11.0+94978d7-32
kubernetes v1.11.0+d4cacc0
features: Basic-Auth GSSAPI Kerberos SPNEGO

Server https://192.168.42.137:8443
kubernetes v1.11.0+d4cacc0

$ ./oc cluster up --public-hostname 192.168.42.137        
Getting a Docker client ...
Checking if image openshift/origin-control-plane:v3.11 is available ...
Pulling image openshift/origin-control-plane:v3.11
E1026 14:31:57.782743    1954 helper.go:173] Reading docker config from /home/docker/.docker/config.json failed: open /home/docker/.docker/config.json: no such file or directory, will attempt to pull image docker.io/openshift/origin-control-plane:v3.11 anonymously
Pulled 1/5 layers, 23% complete
Pulled 2/5 layers, 40% complete
Pulled 3/5 layers, 71% complete
Pulled 4/5 layers, 95% complete
Pulled 5/5 layers, 100% complete
Extracting
Image pull complete
Pulling image openshift/origin-cli:v3.11
E1026 14:32:26.352290    1954 helper.go:173] Reading docker config from /home/docker/.docker/config.json failed: open /home/docker/.docker/config.json: no such file or directory, will attempt to pull image docker.io/openshift/origin-cli:v3.11 anonymously
Image pull complete
Pulling image openshift/origin-node:v3.11
E1026 14:32:32.859355    1954 helper.go:173] Reading docker config from /home/docker/.docker/config.json failed: open /home/docker/.docker/config.json: no such file or directory, will attempt to pull image docker.io/openshift/origin-node:v3.11 anonymously
Pulled 5/6 layers, 88% complete
Pulled 6/6 layers, 100% complete
Extracting
Image pull complete
Checking type of volume mount ...
Determining server IP ...
Using public hostname IP 192.168.42.137 as the host IP
Checking if OpenShift is already running ...
Checking for supported Docker version (=>1.22) ...
Checking if insecured registry is configured properly in Docker ...
Checking if required ports are available ...
Checking if OpenShift client is configured properly ...
Checking if image openshift/origin-control-plane:v3.11 is available ...
Starting OpenShift using openshift/origin-control-plane:v3.11 ...
I1026 14:32:48.836997    1954 config.go:40] Running "create-master-config"
I1026 14:32:50.836266    1954 config.go:46] Running "create-node-config"
I1026 14:32:51.788048    1954 flags.go:30] Running "create-kubelet-flags"
I1026 14:32:52.204661    1954 run_kubelet.go:49] Running "start-kubelet"
I1026 14:32:52.387277    1954 run_self_hosted.go:181] Waiting for the kube-apiserver to be ready ...
I1026 14:33:45.407613    1954 interface.go:26] Installing "kube-proxy" ...
I1026 14:33:45.407917    1954 interface.go:26] Installing "kube-dns" ...
I1026 14:33:45.407924    1954 interface.go:26] Installing "openshift-service-cert-signer-operator" ...
I1026 14:33:45.407929    1954 interface.go:26] Installing "openshift-apiserver" ...
I1026 14:33:45.407958    1954 apply_template.go:81] Installing "openshift-apiserver"
I1026 14:33:45.408147    1954 apply_template.go:81] Installing "kube-dns"
I1026 14:33:45.408148    1954 apply_template.go:81] Installing "kube-proxy"
I1026 14:33:45.409446    1954 apply_template.go:81] Installing "openshift-service-cert-signer-operator"
I1026 14:33:50.708902    1954 interface.go:41] Finished installing "kube-proxy" "kube-dns" "openshift-service-cert-signer-operator" "openshift-apiserver"
I1026 14:35:45.758573    1954 run_self_hosted.go:242] openshift-apiserver available
I1026 14:35:45.758907    1954 interface.go:26] Installing "openshift-controller-manager" ...
I1026 14:35:45.758925    1954 apply_template.go:81] Installing "openshift-controller-manager"
I1026 14:35:49.013333    1954 interface.go:41] Finished installing "openshift-controller-manager"
Adding default OAuthClient redirect URIs ...
Adding web-console ...
Adding registry ...
Adding persistent-volumes ...
Adding centos-imagestreams ...
Adding router ...
Adding sample-templates ...
I1026 14:35:49.033873    1954 interface.go:26] Installing "openshift-web-console-operator" ...
I1026 14:35:49.033882    1954 interface.go:26] Installing "openshift-image-registry" ...
I1026 14:35:49.033888    1954 interface.go:26] Installing "persistent-volumes" ...
I1026 14:35:49.033894    1954 interface.go:26] Installing "centos-imagestreams" ...
I1026 14:35:49.033899    1954 interface.go:26] Installing "openshift-router" ...
I1026 14:35:49.033904    1954 interface.go:26] Installing "sample-templates" ...
I1026 14:35:49.033940    1954 interface.go:26] Installing "sample-templates/mariadb" ...
I1026 14:35:49.033946    1954 interface.go:26] Installing "sample-templates/postgresql" ...
I1026 14:35:49.033951    1954 interface.go:26] Installing "sample-templates/sample pipeline" ...
I1026 14:35:49.033958    1954 interface.go:26] Installing "sample-templates/jenkins pipeline ephemeral" ...
I1026 14:35:49.033963    1954 interface.go:26] Installing "sample-templates/mongodb" ...
I1026 14:35:49.033968    1954 interface.go:26] Installing "sample-templates/mysql" ...
I1026 14:35:49.033973    1954 interface.go:26] Installing "sample-templates/cakephp quickstart" ...
I1026 14:35:49.033978    1954 interface.go:26] Installing "sample-templates/dancer quickstart" ...
I1026 14:35:49.033984    1954 interface.go:26] Installing "sample-templates/django quickstart" ...
I1026 14:35:49.033990    1954 interface.go:26] Installing "sample-templates/nodejs quickstart" ...
I1026 14:35:49.033995    1954 interface.go:26] Installing "sample-templates/rails quickstart" ...
I1026 14:35:49.034034    1954 apply_list.go:67] Installing "sample-templates/rails quickstart"
I1026 14:35:49.034378    1954 apply_template.go:81] Installing "openshift-web-console-operator"
I1026 14:35:49.035238    1954 apply_list.go:67] Installing "centos-imagestreams"
I1026 14:35:49.035607    1954 apply_list.go:67] Installing "sample-templates/mariadb"
I1026 14:35:49.035714    1954 apply_list.go:67] Installing "sample-templates/postgresql"
I1026 14:35:49.035797    1954 apply_list.go:67] Installing "sample-templates/sample pipeline"
I1026 14:35:49.035874    1954 apply_list.go:67] Installing "sample-templates/jenkins pipeline ephemeral"
I1026 14:35:49.035954    1954 apply_list.go:67] Installing "sample-templates/mongodb"
I1026 14:35:49.036029    1954 apply_list.go:67] Installing "sample-templates/mysql"
I1026 14:35:49.036118    1954 apply_list.go:67] Installing "sample-templates/cakephp quickstart"
I1026 14:35:49.036196    1954 apply_list.go:67] Installing "sample-templates/dancer quickstart"
I1026 14:35:49.036270    1954 apply_list.go:67] Installing "sample-templates/django quickstart"
I1026 14:35:49.036351    1954 apply_list.go:67] Installing "sample-templates/nodejs quickstart"
I1026 14:36:01.433171    1954 interface.go:41] Finished installing "sample-templates/mariadb" "sample-templates/postgresql" "sample-templates/sample pipeline" "sample-templates/jenkins pipeline ephemeral" "sample-templates/mongodb" "sample-templates/mysql" "sample-templates/cakephp quickstart" "sample-templates/dancer quickstart" "sample-templates/django quickstart" "sample-templates/nodejs quickstart" "sample-templates/rails quickstart"
I1026 14:36:33.108585    1954 interface.go:41] Finished installing "openshift-web-console-operator" "openshift-image-registry" "persistent-volumes" "centos-imagestreams" "openshift-router" "sample-templates"
Login to server ...
Creating initial project "myproject" ...
Server Information ...
OpenShift server started.

The server is accessible via web console at:
    https://192.168.42.137:8443/console

You are logged in as:
    User:     developer
    Password: <any value>

To login as administrator:
    oc login -u system:admin
```

**Comment by jwforres** [comment]
thanks, will go ahead and
/lgtm

**Comment by nstielau** [comment]
/retest

(althogh I don't think I have perms.  The detailed merge statuses are still from the 26th, so looks like yesterdays retest didn't kick off new Jenkins jobs, might be something that we have to look into with Prow)

**Pull request merged**